docs/book/architecture/translations.rst

```diff
@@ -0,0 +1,125 @@
+.. index::
+   single: Translations
+
+Translations
+============
+
+Sylius uses the approach of personal translations - where each entity is bound with a translation entity, that has it's
+own table (instead of keeping all translations in one table for the whole system).
+This results in having the ``ProductTranslation`` class and ``sylius_product_translation`` table for the ``Product`` entity.
+
+The logic of handling translations in Sylius is in the **ResourceBundle**
+
+The fields of an entity that are meant to be translatable are saved on the translation entity, only their getters and setters
+are also on the original model.
+
+Let's see an example:
+
+Assuming that we would like to have a translatable model of a ``Supplier``, we need a Supplier class and a SupplierTranslation class.
+
+.. code-block:: php
+
+   <?php
+
+   namespace AppBundle\Entity;
+
+   use Sylius\Component\Resource\Model\AbstractTranslation;
+
+   class SupplierTranslation extends AbstractTranslation
+   {
+       /**
+        * @var string
+        */
+       protected $name;
+
+       /**
+        * @return string
+        */
+       public function getName()
+       {
+           return $this->name;
+       }
+
+       /**
+        * @param string $name
+        */
+       public function setName($name)
+       {
+           $this->name = $name;
+       }
+   }
+
+The actual entity has access to its translation by using the ``TranslatableTrait`` which provides the ``getTranslation()`` method.
+
+.. code-block:: php
+
+   <?php
+
+   namespace AppBundle\Entity;
+
+   use Sylius\Component\Resource\Model\TranslatableInterface;
+   use Sylius\Component\Resource\Model\TranslatableTrait;
+
+   class Supplier implements TranslatableInterface
+   {
+       use TranslatableTrait;
```

Comment:
The translations collection needs to be initialised in the supplier's constructor.